File: common/public/src/main/java/dcp/mc/pstp/config/DamageProtection.java
package dcp.mc.pstp.config;

public record DamageProtection(
        boolean projectiles,
        boolean sweeping,
        boolean direct,
        boolean explosions,
        boolean vanillaTeams
)
{
    DamageProtection() {
        this(true, true, true, true, false);
    }
}

File: common/src/main/kotlin/dcp/mc/pstp/config/DamageProtection.kt
package dcp.mc.pstp.config
@JvmRecord
data class DamageProtection(
    val projectiles: Boolean = true,
    val sweeping: Boolean = true,
    val direct: Boolean = true,
    val explosions: Boolean = true,
    val vanillaTeams: Boolean = false,
)